datanode/api/trading_data_v2.go

```diff
@@ -2861,6 +2868,9 @@ func (t *tradingDataServiceV2) ListAllDeHistorySegments(context.Context, *v2.Lis
 func (t *tradingDataServiceV2) FetchDeHistorySegment(ctx context.Context, req *v2.FetchDeHistorySegmentRequest) (*v2.FetchDeHistorySegmentResponse, error) {
 	defer metrics.StartAPIRequestAndTimeGRPC("FetchDeHistorySegment0'")()
 	segment, err := t.deHistoryService.FetchHistorySegment(ctx, req.HistorySegmentId)
+	if t.deHistoryService == nil {
```

Comment:
I think this one needs to be a line higher, above `FetchHistorySegment()`?

Reply:
:)  thanks!